### ISSUE TITLE
Is linear access required?

### ISSUE THREAD

**Issue by mleise** [pinterf/mvtools#11]
I've noticed strange behavior with MFlow inside AvsPmod 2.5.1 x86-64. First of all the error log is filled with this for some frames:
```
Traceback (most recent call last):
  File "avsp.pyo", line 11175, in OnMouseMotionVideoWindow
  File "avsp.pyo", line 13932, in SetVideoStatusText
  File "avsp.pyo", line 14081, in GetVideoInfoDict
  File "avsp.pyo", line 17271, in FormatTime
TypeError: %d format: a number is required, not float
```
MVTools somehow breaks the assumptions AvsPmod makes about the frame times and it is so far the only filter I noticed doing that. But the real trouble I have is with making MFlow yield reproducible results. When playing the script linearly I get different results as with reloading a specific frame. It is reproducible fairly simply with a field separated PAL DV, YV12 clip and the following lines:
```
super = MSuper()
MFlow(super,MAnalyse(super))
```
At the start of a new scene, the linear play trough shows the previous frame distorted by the false motion vectors:
![linear](https://user-images.githubusercontent.com/609447/35118337-bfd8c7b4-fc91-11e7-8067-f467bb6e8351.png)
A reload on the other hand shows the original frame at that point in time (as if it suddenly detected a scene change and used that as a fallback):
![reload](https://user-images.githubusercontent.com/609447/35118338-bffa9cfe-fc91-11e7-86bb-671d7a59cb7c.png)
One thing I noticed is that the line pitch of the garbled frames is 768 bytes, while the reloaded frames have a pitch of 1536. In fact after this reload when I go to the frame right before the scene change it also shows the original image (without MFlow applied) at a pitch of 1536. Another reload and it is back to 768 bytes with MFlow applied. Since I use forward prediction, this frame should not be affected by the scene change though:
![previous frame without reload](https://user-images.githubusercontent.com/609447/35119065-1409fc84-fc94-11e7-89af-78dc17353899.png)
![previous frame after reload](https://user-images.githubusercontent.com/609447/35119063-13dfa61e-fc94-11e7-9c81-28d848c47333.png)

Setting a very high thSCD1 removes the discrepancy: Both the frame before and after the scene change are always extrapolated. That suggests that scene changes are not detected on linear playback where they should be. Also I'm wondering if the previous frame should really be affected by the scene change since it doesn't depend on any motion vectors that would cross scenes.

**Comment by pinterf** [comment]
Thanks for the report, I'll see into it later. Just writing that I noticed this post. (having a short look into the source I couldn't see any reason for doing this, but mvtools internals are never easy to understand so I may have to dig into it)

**Comment by mleise** [comment]
Yeah I figured so much, since you are maintaining code that someone else wrote. A burden I never took onto myself. 👍

**Comment by pinterf** [comment]
Fixed. See it probably from a future 2.7.31-. This bug was fixed in upstream 2.5.11.22 but pfmod branch was based on an earlier "fork", and this fix was missed during the manual merge. Now it's done. 
https://github.com/pinterf/mvtools/commit/d6bfc4d6e0fb670c7a2aad27ed4268124a7714b3

**Comment by mleise** [comment]
Glad it didn't take a lot of effort then ;) thx
